File: virtual-queue/src/shared/format/counterLabel.ts
import type { CounterLabelMode } from '../types/api'

/** Maps counter index to display label according to place mode. */
export function counterLabel(
  n: number | null | undefined,
  mode: CounterLabelMode = 'LETTERS',
): string {
  if (n == null || n < 1) return '—'
  if (mode === 'NUMBERS') return String(n)
  if (n <= 26) return String.fromCharCode(64 + n)
  return String(n)
}

File: virtual-queue-mobile/lib/core/format/counter_label.dart
/// Maps counter index 1 → A, 2 → B, …
String counterLabel(int? n) {
  if (n == null || n < 1) return '';
  if (n <= 26) {
    return String.fromCharCode(64 + n);
  }
  return '$n';
}

String counterDisplay(int? number, [String? label]) {
  final code = (label != null && label.isNotEmpty) ? label : counterLabel(number);
  return code;
}
